app/ui/src/main/java/com/fsck/k9/ui/settings/SettingsActivity.kt

```diff
@@ -1,108 +1,189 @@
         }
+
+        return true
     }
 
-    private fun populateSettingsList(accounts: List<Account>) {
-        settingsAdapter.clear()
+    inner class TabAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
+        private val generalSettings = GeneralSettingsTab()
+        private val accountSettings = AccountSettingsTab()
 
-        val miscSection = Section().apply {
-            val item = SettingsActionItem(getString(R.string.about_action), SettingsAction.ABOUT_SCREEN, R.attr.iconSettingsAbout)
-            add(item)
+        override fun getItem(i: Int): Fragment? = when (i) {
+            0 -> generalSettings
+            1 -> accountSettings
+            else -> null
         }
-        settingsAdapter.add(miscSection)
 
-        val generalSection = Section().apply {
-            val item = SettingsActionItem(getString(R.string.general_settings_title),
-                    SettingsAction.GENERAL_SETTINGS, R.attr.iconSettingsGeneral)
-            add(item)
+        override fun getPageTitle(i: Int): CharSequence? = when (i) {
+            0 -> getString(R.string.general_settings_title)
+            1 -> getString(R.string.accounts_title)
+            else -> null
         }
-        settingsAdapter.add(generalSection)
 
-        val accountSection = Section().apply {
-            for (account in accounts) {
-                add(AccountItem(account))
-            }
-            val item = SettingsActionItem(getString(R.string.add_account_action), SettingsAction.ADD_ACCOUNT, R.attr.iconSettingsAccountAdd)
-            add(item)
+        override fun getCount(): Int = 2
+    }
+
+    internal class GeneralSettingsTab : Fragment() {
+        override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+            return inflater.inflate(R.layout.general_settings_tab, container, false)
         }
-        accountSection.setHeader(SettingsDividerItem(getString(R.string.accounts_title)))
-        settingsAdapter.add(accountSection)
     }
 
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        return when (item.itemId) {
-            android.R.id.home -> {
-                onBackPressed()
-                true
+    internal class AccountSettingsTab : Fragment(), AdapterView.OnItemSelectedListener {
+        override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+            return inflater.inflate(R.layout.account_settings, container, false)
+        }
+
+        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+            clearAccounts()
+
+            val prefs = Preferences.getPreferences(context)
+
+            val viewModel: SettingsViewModel by viewModel()
+            viewModel.accounts.observeNotNull(this) {
+                loadAccounts(prefs.getAccounts())
```

Comment:
I hit an issue when trying to use this observer the proper way, if the default activity is the accounts page, and you
1. Launch app
2. Open settings
3. Go back to accounts activity
4. Delete the first account
5. Re-open settings

The observer runs twice after re-opening
The first time it gives the list before account removal, in the second it has been removed
Is there a better way to prevent or work around this issue?